app/src/main/java/com/example/keniphhomemaintenance/dwellings/DwellingsScreenView.kt

```diff
@@ -0,0 +1,226 @@
+                }
+            )
+        }
+
+    }
+}
+
+@ExperimentalComposeUiApi
+@Composable
+fun NewDwellingDialog(
+    modifier: Modifier = Modifier,
+    newName: MutableState<String> = remember { mutableStateOf("") },
+    newAddress: MutableState<String> = remember { mutableStateOf("") },
+) {
+    var expanded by remember { mutableStateOf(false) }
+    val focusRequester = remember { FocusRequester() }
+    val keyboardController = LocalSoftwareKeyboardController.current
+
+
+    val spinnerList: MutableList<DwellingType> = mutableListOf()
+    for (item in DwellingType.values()) {
+        spinnerList.add(item)
+    }
+    var selectedSpinnerItem by remember { mutableStateOf("") }
+    var selectedSpinnerItemLabel: DwellingType by remember { mutableStateOf(DwellingType.Other) }
+    var spinnerTextFieldSize by remember { mutableStateOf(Size.Zero) }
+    val spinnerIcon = if (expanded) {
+        Icons.Filled.KeyboardArrowUp
+    } else {
+        Icons.Filled.KeyboardArrowDown
+    }
+
+    Dialog(onDismissRequest = { addDwellingDialog.value = false })
+    {
+        Column(
+            modifier
+                .wrapContentSize()
+                .background(Color.LightGray)
+                .padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center,
+        ) {
+            Text(text = "New dwelling!", fontSize = 20.sp, textAlign = TextAlign.Center)
+            Divider()
+
+            OutlinedTextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .focusRequester(focusRequester),
+                value = newName.value,
+                onValueChange = { newName.value = it },
+                label = { Text("Dwelling name") },
+                keyboardOptions = KeyboardOptions(
+                    keyboardType = KeyboardType.Text,
+                    imeAction = ImeAction.Next
+                ),
+                keyboardActions = KeyboardActions(onNext = {
+                    keyboardController?.hide()
+                    focusRequester.requestFocus()
+                })
+            )
+
+            OutlinedTextField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .focusRequester(focusRequester),
+                value = newAddress.value,
+                onValueChange = { newAddress.value = it },
+                label = { Text("Dwelling address") },
+                keyboardOptions = KeyboardOptions(
+                    keyboardType = KeyboardType.Text,
+                    imeAction = ImeAction.Done
+                ),
+                keyboardActions = KeyboardActions(onDone = {
+                    keyboardController?.hide()
+                    focusRequester.requestFocus()
+                })
+            )
+
+            Column(modifier = Modifier.padding(20.dp)) {
+
+                OutlinedTextField(
```

Comment:
Let's remove this for now and not have the user specify a dwelling type. Keeps with what we chatted about the other day - not complicating the base functionality before we need to. We won't be handling different dwelling types for a while, let's not get ahead of ourselves.